protocol/test/foundry/silo/silo.t.sol

```diff
@@ -0,0 +1,261 @@
+        int96[] memory stems = new int96[](transfers);
+        uint256[] memory amounts = new uint256[](transfers);
+        for (uint256 i; i < transfers; i++) {
+            stems[i] = int96(uint96(i * stalkEarnedPerSeason));
+            amounts[i] = depositAmount;
+        }
+        bs.transferDeposits(farmers[0], farmers[1], C.BEAN, stems, amounts);
+        vm.stopPrank();
+
+        depositIds = bs.getTokenDepositIdsForAccount(farmers[0], C.BEAN);
+        deposit = bs.getTokenDepositsForAccount(farmers[0], C.BEAN);
+        verifyDepositIdLengths(depositIds, deposit, C.BEAN, deposits - transfers);
+
+        depositIds = bs.getTokenDepositIdsForAccount(farmers[1], C.BEAN);
+        deposit = bs.getTokenDepositsForAccount(farmers[1], C.BEAN);
+        verifyDepositIdLengths(depositIds, deposit, C.BEAN, transfers);
+
+        //////////// CONVERT ////////////
+
+        // increase deltaB.
+        addLiquidityToWell(C.BEAN_ETH_WELL, 0, 100 ether);
+        // skip germination as germinating deposits cannot be converted.
+        bs.siloSunrise(0);
+        bs.siloSunrise(0);
+
+        // converts deposits for farmer[1].
+        vm.startPrank(farmers[1]);
+        bs.convert(
+            createBeanToLPConvert(C.BEAN_ETH_WELL, depositAmount * transfers),
+            stems,
+            amounts
+        );
+        vm.stopPrank();
+
+        depositIds = bs.getTokenDepositIdsForAccount(farmers[1], C.BEAN);
+        deposit = bs.getTokenDepositsForAccount(farmers[1], C.BEAN);
+        verifyDepositIdLengths(depositIds, deposit, C.BEAN, 0);
+
+        depositIds = bs.getTokenDepositIdsForAccount(farmers[1], C.BEAN_ETH_WELL);
+        deposit = bs.getTokenDepositsForAccount(farmers[1], C.BEAN_ETH_WELL);
+        verifyDepositIdLengths(depositIds, deposit, C.BEAN_ETH_WELL, 1);
+
+        //////////// WITHDRAW ////////////
+
+        // withdraws deposits from farmer[0]
+        stems = new int96[](deposits - transfers);
+        amounts = new uint256[](deposits - transfers);
+        for (uint256 i; i < deposits - transfers; i++) {
+            stems[i] = int96(uint96((i + transfers) * stalkEarnedPerSeason));
+            amounts[i] = depositAmount;
+        }
+        vm.startPrank(farmers[0]);
+        bs.withdrawDeposits(C.BEAN, stems, amounts, 0);
+        depositIds = bs.getTokenDepositIdsForAccount(farmers[0], C.BEAN);
+        deposit = bs.getTokenDepositsForAccount(farmers[0], C.BEAN);
+        verifyDepositIdLengths(depositIds, deposit, C.BEAN, 0);
+    }
+
+    // silo list helpers //
+    function verifyDepositIdLengths(
+        uint256[] memory depositIds,
+        IMockFBeanstalk.TokenDepositId memory deposit,
+        address token,
+        uint256 length
+    ) internal {
+        assertEq(depositIds.length, length);
+        assertEq(depositIds.length, deposit.depositIds.length);
+        assertEq(depositIds.length, deposit.tokenDeposits.length);
+        assertEq(deposit.token, token);
+    }
```

Comment:
decided against reversions here as they would fall in the scope of general field/silo/market tests - adding the index doesn't add additional restrictions on the user side so it would be slightly redundant here.